framework/src/auxkernels/ElementIntegerAux.C

```diff
@@ -0,0 +1,35 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ElementIntegerAux.h"
+
+registerMooseObject("MooseApp", ElementIntegerAux);
+
+template <>
+InputParameters
+validParams<ElementIntegerAux>()
+{
+  InputParameters params = validParams<AuxKernel>();
+  params.addClassDescription("Creates a field showing the element integer.");
+  params.addParam<std::vector<std::string>>("integer_names", "Element integers to be retrieved");
```

Comment:
It would be good to make `std::vector<std::string>` here just be `ExtraElementIDs`.  See how it's done in `MooseTypes`.

Reply:
Ok. Is `ExtraElementIDNames` better?

Reply:
We have to have `std::vector<ExtraElementIDName>` here because MooseTypes.h only has `DerivativeStringClass`. But I think it is better than raw `std::string` already.